components/membership/membership-rest-impl/src/main/kotlin/net/corda/membership/impl/rest/v1/MemberLookupRestResourceImpl.kt

```
@@ -184,5 +198,24 @@ class MemberLookupRestResourceImpl @Activate constructor(
         }
 
         private fun GroupParameters.toMap() = entries.associate { it.key to it.value }
+
+        private fun List<String>.getStatusFilter(isMgm: Boolean): List<String> {
+            val filter = this.map {
+                val status = it.uppercase()
+                if (!listOf(MEMBER_STATUS_ACTIVE, MEMBER_STATUS_SUSPENDED).contains(status)) {
+                    throw ResourceNotFoundException("Invalid status: $it")
+                }
+                status
+            }
+
+            if (filter.isEmpty()) {
+                return listOf(MEMBER_STATUS_ACTIVE)
+            }
+
+            return filter.filter { if (!isMgm) (it in setOf(MEMBER_STATUS_ACTIVE)) else true }.distinct()
+        }
+
+        private fun MembershipGroupReader.isMgm(holdingIdentity: HoldingIdentity): Boolean =
+            lookup(holdingIdentity.x500Name)?.isMgm ?: false
```

Shouldn't we be failing here if the vnode is not present in the members list, instead of assuming it's a regular vnode?

---

The only case where this would be null is before registering, and at that point we need to be able to see the MGM details. Even if we _somehow_ end up in a weird scenario where a non-MGM vnode can't see itself in the member list, that vnode's member list won't have suspended members since they are not distributed, so there won't be risk of returning suspended members from here

---

hm, I see. I think that's not ideal. It feels like this information should be coming from the virtual node metadata, instead of having to perform a member lookup. But, I understand this is not possible at the moment without a lot of extra work.  So, we'll have to live with this for now. I created a [ticket](https://r3-cev.atlassian.net/browse/CORE-11226) for us to address this later on. 

I think it's worth adding a comment there explaining the reason for the defaulting and adding a reference to the ticket.